xarray/test/test_dataarray.py

```diff
@@ -2283,23 +2283,20 @@ def test_binary_op_join_setting(self):
                            reshape(len(xcoords),len(ycoords),len(zcoords)),
                            [(xdim, xcoords), (ydim, ycoords),(zdim, zcoords)])
         # now create a data array with the last x slice missing
-        arr1 = arr[0:-1,:,:].copy()
+        missing_last_x = arr[0:-1,:,:].copy()
         # create another data array with the last z slice missing
-        arr2 = arr[:,:,0:-1].copy()
+        missing_last_z = arr[:,:,0:-1].copy()
         # because the default in OPTIONS is join="inner", we test "outer" first
         xr.set_options(join="outer")
-        result = arr1 + arr2
-        self.assertTrue(result.size == total_size) # should be 3 * 3 * 3
+        result = missing_last_x + missing_last_z
         self.assertTrue(result.shape == arr.shape)
         self.assertTrue(result[-1,:,:].isnull().all())
         self.assertTrue(result[:,:,-1].isnull().all())
         # now revert back to join="inner"
         xr.set_options(join="inner")
-        result = arr1 + arr2
-        self.assertTrue(result.size == \
-                        (len(xcoords)-1)*len(ycoords)*(len(zcoords)-1))
+        result = missing_last_x + missing_last_z
         self.assertTrue(result.shape == \
                         (len(xcoords)-1, len(ycoords), len(zcoords)-1))
         self.assertTrue(result.notnull().all())
-        self.assertTrue('c' not in list(result['x']))
-        self.assertTrue(2 not in list(result['z']))
+        self.assertFalse('c' in list(result['x']))
```

Comment:
What's special about '2' and 'c'? (I know these are actually the x and z components that get dropped.) For readability, you could set those between lines 2278 and 2279, though. i.e.:
```
x_drop = 'c'
z_drop = 2
xcoords, ycoords, zcoords = ['a', 'b', x_drop], [-2, 0, 2], [0, 1, z_drop]
```
and then use them to make super explicit what's going on in the test/assertion:
```
self.assertFalse(x_drop in list(result['x']))
etc.
```